### PR TITLE
feature/GDEV-635

### PR DESCRIPTION
**Pull request by ac-jorellanaf**
Implemented dataset details layout with some necessary placeholders (cannot be addressed now). Made room for data request button. Minor expansion to dataset models. Renamed Dataset to DatasetDetails and dataset folder to datasetlist.

### CODE REVIEW

File: src/components/browse/datasetlist/components/datasetDetailsLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
```
Comment (alamgirmani) [inline src/components/browse/datasetlist/components/datasetDetailsLayout.tsx:1]: The component is the top level folder name. Do not use components further down the structure. Follow the same structure we follow inside the components for nesting. Browse is one of the components and it contains sidebar and datasetList. So datasetList will have its own file and folders and so on..

Comment (ac-jorellanaf) [reply src/components/browse/datasetlist/components/datasetDetailsLayout.tsx]: Done

File: src/components/browse/datasetlist/datasetheader/datasetdetails/datasetDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <div className="fs-8">
+      <Row>
+        <Col lg md sm xl xs xxl="10">
+          <p className="my-0">
+            <span className="fw-bold">Dataset ID:&nbsp;</span>
+            {hit.content.accession}
+
+            <br />
+            <span className="fw-bold">Full title:&nbsp;</span>
```
Comment (alamgirmani) [inline src/components/browse/datasetlist/datasetheader/datasetdetails/datasetDetails.tsx:25]: This field is not in the wireframes. We don't need it.

File: src/components/browse/datasetlist/datasetheader/datasetdetails/datasetFiles.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { FileEarmark } from "react-bootstrap-icons";
+
+interface dataSetFilesProps {
+  filesList: string[];
+}
+
+const DatasetFiles = (props: dataSetFilesProps) => {
+  return (
+    <DatasetDetailsLayout
+      icon={<FileEarmark size={32} />}
```
Comment (alamgirmani) [inline src/components/browse/datasetlist/datasetheader/datasetdetails/datasetFiles.tsx:13]: Lets go with the grey from the wireframes. I checked with Koray and he would like to stick with the design.

Comment (ac-jorellanaf) [reply src/components/browse/datasetlist/datasetheader/datasetdetails/datasetFiles.tsx]: Updated.

**Review by alamgirmani — APPROVED** [review]

**Pull request merged**